lib/stun/stun.go

```diff
@@ -100,24 +103,25 @@ func New(cfg config.Wrapper, subscriber Subscriber, conn net.PacketConn) (*Servi
 
 		natType: NATUnknown,
 		addr:    nil,
-		stop:    make(chan struct{}),
-	}, otherDataConn
+	}
+	s.Service = util.AsService(s.serve)
+	return s, otherDataConn
 }
 
 func (s *Service) Stop() {
-	close(s.stop)
+	s.Service.Stop()
```

Comment:
Again, can the stunConn be closed in serve?

Reply:
In this case it looks like no, as the `stunConn` is established in `New`, not in `serve`.